src/main/java/duke/core/TypoCorrector.java

```diff
@@ -0,0 +1,115 @@
+package duke.core;
+
+import org.apache.commons.text.similarity.LevenshteinDistance;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * This is a command typo corrector for Duke user command.
+ * It provides a method TypoCorrector.CommandCorrection which takes in an invalid input command
+ * and return a possible matched command.
+ *
+ * @author HUANG XUAN KUN
+ * @version 1.2
+ */
+public class TypoCorrector {
+
+    //The maximum ratio changes of a text in % that is acceptable
+    private static final double MAX_DISTANCE_DIFF_RATIO = 0.5;
+
+    //Sets of "Dictionaries" for the command keyword, categorised by number of keywords contain in a supported commands.
+    private static final ArrayList<String> oneKeywordCommand = new ArrayList<String>(
+            Arrays.asList("bye"));
+    private static final ArrayList<String> twoKeywordsCommands = new ArrayList<String>(
+            Arrays.asList("list patients", "list tasks"));
+    private static final ArrayList<String> otherCommands = new ArrayList<String>(
+            Arrays.asList("update patient", "update task",
+                    "delete patient", "delete task", "add task", "add patient",
+                    "assign by", "find patient", "find task"));
+
+    /**
+     * This method take in an user input command with typo and return a possible matches
+     * If the return string is equal to the input command, there is no match.
+     *
+     * @param command the full userInput command without parsing
+     * @return a string of correctedCommand
+     */
+    public static String commandCorrection(String command) {
+        String[] splitCommand = command.split("\\s+");
+        int commandSize = splitCommand.length;
+        String closestMatch;
+        String firstTwoKeywords;
+        if (commandSize == 1 || command.length() <= 6) {
```

Comment:
Are you taking the greedy approach to checking commands when verifying the command length is <= 6? I think this might make your check a little excessive/might make future one-word command additions more difficult to implement if they're over 6 characters long.

Reply:
Yes you are right.🤣It is definitely excessive for now as the only extended coverage will be cases like `b ye`, `sav e` (possible). 
When there is lengthy one-word command being implemented, such as `frequent`, I will remove the condition check for <=6.